In [18]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

In [19]:
mean_new_drug = 10 #mean improvement for new drug
mean_placebo = 5 #mean improvement for placebo group
std_new_drug = 5 #std for new drug group
std_placebo = 5 #std for placebo group
n_new_drug=50 #sample size for new drug group
n_placebo=50 #sample size for placebo group

#calculate the z score
z= (mean_new_drug -mean_placebo)/ np.sqrt(
    (std_new_drug**2 /n_new_drug) + (std_placebo **2 / n_placebo))

# calculate the p value(two tailed test)
p_value = 2*(1- stats.norm.cdf(abs(z)))

#output the results
print(f"Z-score:{z}")
print(f"P-value:{p_value}")

#decision based on a significance level(alpha)
alpha = 0.05
if p_value< alpha:
    print("Reject the null hypothesis(there is a significant difference between the values)")
else:
    print("Fail to reject the null hypothesis(no significant difference between the values)")

Z-score:5.0
P-value:5.733031438470704e-07
Reject the null hypothesis(there is a significant difference between the values)


In [20]:
#sample data - average population weight in kgs
data= [78,62,80,75,61,74,79,93,77,95,67,88,54,69]

In [21]:
#define the null hypothesis(average population weight= 80)
null_mean = 80

In [22]:
len(data)

14

In [23]:
#calculate the sample mean and std
sample_mean= np.mean(data)
sample_std= np.std(data)

In [24]:
sample_mean

75.14285714285714

In [25]:
sample_std

11.506874608915581

In [26]:
#calculate the z-score
z_score=(sample_mean - null_mean)/sample_std
z_score

-0.4221079156784696

In [27]:
p_value= 2*(1- stats.norm.cdf(abs(z_score)))
p_value

0.6729462504556214

In [28]:
alpha = 0.05
if p_value< alpha:
    print("Reject the null hypothesis(there is a significant difference between the values)")
else:
    print("Fail to reject the null hypothesis(no significant difference between the values)")

Fail to reject the null hypothesis(no significant difference between the values)


In [34]:
from statsmodels.stats.weightstats import ztest

In [35]:
# enter IQ levels for 20 patient
data = [88,92,94,94,96,97,97,97,99,99,105,109,109,109,110,112,112,113,114,115]

In [36]:
# Perform one sample z-test
ztest(data, value=100)

(1.5976240527147705, 0.1101266701438426)

##### The test statistic for the one sample z-test is 1.5976 & the corresponding p-value is 0.1101.

##### Since this p-value is not less than 0.5, we dont have sufficient evidence to reject the null hypothesis.
##### In other words, the new drug does not significantly affect IQ level.

## CHI SQAURE TEST 

In [38]:
from scipy.stats import chi2_contingency
import numpy as np

In [39]:
data = np.array([[207,22,241] , [234,242,232]])

In [40]:
data

array([[207,  22, 241],
       [234, 242, 232]])

In [41]:
data.sum(axis=0)

array([441, 264, 473])

In [42]:
data.sum(axis=1)

array([470, 708])

In [43]:
stat, p, dof, expected = chi2_contingency(data)

In [44]:
stat

142.90604727381384

In [45]:
dof

2

In [46]:
expected

array([[175.95076401, 105.33106961, 188.71816638],
       [265.04923599, 158.66893039, 284.28183362]])

In [49]:
alpha = 0.05
print("p value is " + str(p))
if p <= alpha:
   print ('Dependent (reject H0)')
else:
   print("Independent (H0 holds true)")

p value is 9.29707039184022e-32
Dependent (reject H0)


In [62]:
data = np.array([[5,10,5] , [5,15,0],[5,10,15]])

In [63]:
data

array([[ 5, 10,  5],
       [ 5, 15,  0],
       [ 5, 10, 15]])

In [64]:
data.sum(axis=0)

array([15, 35, 20])

In [65]:
data.sum(axis=1)

array([20, 20, 30])

In [66]:
stat, p, dof, expected = chi2_contingency(data)

In [67]:
stat

15.347222222222225

In [68]:
p

0.0040326727858557955

In [69]:
dof

4

In [70]:
expected

array([[ 4.28571429, 10.        ,  5.71428571],
       [ 4.28571429, 10.        ,  5.71428571],
       [ 6.42857143, 15.        ,  8.57142857]])

In [71]:
alpha = 0.05
print("p value is " + str(p))
if p <= alpha:
   print ('Dependent (reject H0)')
else:
   print("Independent (H0 holds true)")

p value is 0.0040326727858557955
Dependent (reject H0)


In [72]:
data = np.array([[15,5,5] , [10,10,15],[2,3,5]])

In [73]:
data

array([[15,  5,  5],
       [10, 10, 15],
       [ 2,  3,  5]])

In [74]:
data.sum(axis=0)

array([27, 18, 25])

In [75]:
data.sum(axis=1)

array([25, 35, 10])

In [76]:
stats, p, dof, expected = chi2_contingency(data)

In [77]:
stats

8.07777777777778

In [78]:
p

0.08877028710376675

In [79]:
dof

4

In [80]:
expected

array([[ 9.64285714,  6.42857143,  8.92857143],
       [13.5       ,  9.        , 12.5       ],
       [ 3.85714286,  2.57142857,  3.57142857]])

In [81]:
alpha = 0.05
print("p value is " + str(p))
if p <= alpha:
   print ('Dependent (reject H0)')
else:
   print("Independent (H0 holds true)")

p value is 0.08877028710376675
Independent (H0 holds true)


In [83]:
data = np.array ([[20,5],[10,15]])

In [84]:
data

array([[20,  5],
       [10, 15]])

In [85]:
data.sum(axis=0)

array([30, 20])

In [86]:
data.sum(axis=1)

array([25, 25])

In [87]:
stats, p, dof, expected = chi2_contingency(data)

In [88]:
stats

6.75

In [89]:
p

0.0093747684594349

In [90]:
dof

1

In [91]:
expected

array([[15., 10.],
       [15., 10.]])

In [92]:
alpha = 0.05
print("p value is " + str(p))
if p <= alpha:
   print ('Dependent (reject H0)')
else:
   print("Independent (H0 holds true)")

p value is 0.0093747684594349
Dependent (reject H0)
